presto-hudi/src/main/java/com/facebook/presto/hudi/HudiMetadataFactory.java

```diff
@@ -28,20 +30,30 @@
     private final ExtendedHiveMetastore metastore;
     private final HdfsEnvironment hdfsEnvironment;
     private final TypeManager typeManager;
+    private final long perTransactionCacheMaximumSize;
+    private final boolean metastoreImpersonationEnabled;
+    private final int metastorePartitionCacheMaxColumnCount;
 
     @Inject
     public HudiMetadataFactory(
             ExtendedHiveMetastore metastore,
             HdfsEnvironment hdfsEnvironment,
-            TypeManager typeManager)
+            TypeManager typeManager,
+            MetastoreClientConfig metastoreClientConfig)
     {
         this.metastore = requireNonNull(metastore, "metastore is null");
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.perTransactionCacheMaximumSize = metastoreClientConfig.getPerTransactionMetastoreCacheMaximumSize();
+        this.metastoreImpersonationEnabled = metastoreClientConfig.isMetastoreImpersonationEnabled();
+        this.metastorePartitionCacheMaxColumnCount = metastoreClientConfig.getPartitionCacheColumnCountLimit();
     }
 
     public ConnectorMetadata create()
     {
-        return new HudiMetadata(metastore, hdfsEnvironment, typeManager);
+        return new HudiMetadata(
```

Comment:
Whats the TTL for this cached data? and how is it invalidated?, can you explain for e.g how a stale cache would affect queries? Would it not reflect a partition that was added?

Reply:
Kept the defaults same as hive connector https://prestodb.io/docs/current/connector/hive.html#metastore-configuration-properties and the default TTL is 0s and refreshed every 1ms. The `CachingHiveMetastore` is just a thread-safe `LoadingCache` wrapper on the delegate metastore. If the partition is evicted or not present in cache, delegate will fetch the partition from metastore. For stale cache, the caching metastore also provides a config for refresh interval and refreshes the cache asynchronously. So, it can be configured depending on the workload.